src/components/data_comps/datm/nuopc/datm_comp_mod.F90

```diff
@@ -1426,4 +1422,227 @@ subroutine datm_comp_run(x2a, a2x, &
 
   end subroutine datm_comp_run
 
+  !===============================================================================
+
+  subroutine datm_comp_import(importState, rc)
+
+    ! input/output variables
+    type(ESMF_State)     :: importState
+    integer, intent(out) :: rc
+
+    ! local variables
+    integer :: k
+    !----------------------------------------------------------------
+
+    k = mct_aVect_indexRA(x2a, 'Sx_avsdr')
```

Comment:
If we put the strings into an array 
'Sx_avsdr', 'Sx_avsdf', ...
couldn't this be a simple 3 line loop replacing lines 1437 through 1507?